ui/components/Connect/connections/helm-connect/StepperContent.js

```diff
@@ -0,0 +1,369 @@
+  const formRef = React.createRef();
+  const formStateRef = React.createRef();
+  const [schema, setSchema] = React.useState({});
+  const [isSchemaFetched, setIsSchemaFetched] = React.useState(false);
+  const [formData, setFormData] = React.useState({});
+
+  const handleCallback = () => {
+    if (formRef.current && formRef.current.validateForm()) {
+      // Check if the URL is valid via an API 
+      axios
+        .post(`/api/integrations/connections/helm/verify`, { url: formData.url })
+        .then((response) => {
+          if (response.data.isValid) {
+            // URL is valid, allow the user to proceed
+            handleNext();
+          } else {
+            // URL is not valid, show an error or update the form's error state
+            // You can display an error message or update the form to indicate the URL is invalid
+            // For example, you can use formRef to set a specific field's error message:
+            formRef.current.setError('url', {
+              type: 'custom',
+              message: 'Invalid URL. Please check the URL and try again.',
+            });
+          }
+        })
+        .catch((error) => {
+          // Handle API request error, you can show an error message to the user
+          console.error('Error validating URL:', error);
+        });
+    }
+  };
+  
+  console.log('formRef', formRef);
+
+  const cancelCallback = () => {
+    Router.push('/dashboard');
+  };
+  const handleChange = (data) => {
+    formStateRef.current = data;
+    setFormData(data);
+  };
+
+  React.useEffect(() => {
+    let isMounted = true;
+    VerifyConnectionKind('helm')
+      .then(() => setIsConnected(true))
+      .catch(() => setIsConnected(false));
+
+    const fetchData = async () => {
+      try {
+        const response = await promisifiedDataFetch(`/api/schema/resource/helmRepo`, {
+          method: 'GET',
+          credentials: 'include',
+        });
+
+        // Check if the component is still mounted before setting the state
+        if (isMounted) {
+          setSchema(response);
+          setIsSchemaFetched(true);
+        }
+      } catch (error) {
+        console.error('Error fetching schema in connection wizard', error);
+      }
+    };
+
+    fetchData();
+
+    // Cleanup function
+    return () => {
+      isMounted = false;
+    };
+  }, []);
+
+  return (
+    <StepperContent
+      {...ConnectAppContent}
+      handleCallback={handleCallback}
+      cancelCallback={cancelCallback}
+      disabled={Object.keys(formData).length !== 3}
+      btnText={isConnected ? 'Update' : 'Connect'}
+    >
+      {isSchemaFetched && (
+        <RJSFWrapper
+          key="connect-helm-repo-rjsf-form"
+          formData={formData}
+          onChange={handleChange}
+          uiSchema={schema.uiSchema}
+          jsonSchema={schema.rjsfSchema}
+          liveValidate={false}
+          formRef={formRef}
+          // setErrors={setErrors}
```

Comment:
Missed by mistake?

Reply:
No it was intentional, though we can use extraErrors prop of RJSF to do async validation